systemtest/src/main/java/io/strimzi/systemtest/kafkaclients/externalClients/OauthKafkaClient.java

```diff
@@ -4,29 +4,257 @@
  */
 package io.strimzi.systemtest.kafkaclients.externalClients;
 
+import io.strimzi.systemtest.Constants;
 import io.strimzi.systemtest.kafkaclients.IKafkaClient;
+import io.strimzi.systemtest.resources.crd.KafkaResource;
+import io.strimzi.test.executor.Exec;
+import io.vertx.core.Vertx;
+import org.apache.kafka.clients.CommonClientConfigs;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
 
+import java.io.File;
+import java.io.IOException;
+import java.nio.charset.StandardCharsets;
+import java.nio.file.Files;
+import java.util.Properties;
+import java.util.concurrent.CompletableFuture;
 import java.util.concurrent.Future;
+import java.util.concurrent.TimeUnit;
+import java.util.function.IntPredicate;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
-public class OauthKafkaClient implements IKafkaClient {
+import static io.strimzi.systemtest.kafkaclients.externalClients.KafkaClientProperties.fixBadlyImportedAuthzSettings;
+import static io.strimzi.test.k8s.KubeClusterResource.kubeClient;
+
+public class OauthKafkaClient implements IKafkaClient<Future<Integer>> {
```

Comment:
Can you add a Javadoc to explain what this class is for, and how it differs other implementations of `IKafkaClient`.